finetune/base_models/gpt2/featurizer.py

```diff
@@ -0,0 +1,213 @@
+import functools
```

Comment:
There is a lot of duplicate code between here and the gpt featuriser.

Reply:
Ripped out the exact duplicates now